Formula/elasticsearch-full@7.17.rb

```diff
@@ -0,0 +1,130 @@
+# Release notes: https://www.elastic.co/guide/en/elasticsearch/reference/7.17/es-release-notes.html
+class ElasticsearchFullAT717 < Formula
+    desc "Distributed search & analytics engine"
+    homepage "https://www.elastic.co/products/elasticsearch"
+    url "https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-7.17.14-darwin-x86_64.tar.gz"
+    version "7.17.14"
+    sha256 "3dc253b91a3fc984e2bdaaa43f64ae3844c8dfebd0cd30ab59756a887fcbea74"
+    keg_only :versioned_formula
+  
+    def cluster_name
+      "elasticsearch_#{ENV["USER"]}"
+    end
+  
+    def install
+      # Install everything else into package directory
+      libexec.install "bin", "config", "jdk.app", "lib", "modules"
+  
+      inreplace libexec/"bin/elasticsearch-env",
+                "if [ -z \"$ES_PATH_CONF\" ]; then ES_PATH_CONF=\"$ES_HOME\"/config; fi",
+                "if [ -z \"$ES_PATH_CONF\" ]; then ES_PATH_CONF=\"#{etc}/elasticsearch\"; fi"
+  
+      # Set up Elasticsearch for local development:
+      inreplace "#{libexec}/config/elasticsearch.yml" do |s|
+        # 1. Give the cluster a unique name
+        s.gsub!(/#\s*cluster\.name: .*/, "cluster.name: #{cluster_name}")
+  
+        # 2. Configure paths
+        s.sub!(%r{#\s*path\.data: /path/to.+$}, "path.data: #{var}/lib/elasticsearch/")
+        s.sub!(%r{#\s*path\.logs: /path/to.+$}, "path.logs: #{var}/log/elasticsearch/")
+      end
+  
+      inreplace "#{libexec}/config/jvm.options", %r{logs/gc.log}, "#{var}/log/elasticsearch/gc.log"
+  
+      # Move config files into etc
+      (etc/"elasticsearch").install Dir[libexec/"config/*"]
+      (libexec/"config").rmtree
+  
+      Dir.foreach(libexec/"bin") do |f|
+        next if f == "." || f == ".." || !File.extname(f).empty?
+  
+        bin.install libexec/"bin"/f
+      end
+      bin.env_script_all_files(libexec/"bin", {})
+  
+      system "codesign", "-f", "-s", "-", libexec/"modules/x-pack-ml/platform/darwin-x86_64/controller.app", "--deep"
+      system "find", libexec/"jdk.app/Contents/Home/bin", "-type", "f", "-exec", "codesign", "-f", "-s", "-", "{}", ";"
+    end
+  
+    def post_install
+      # Make sure runtime directories exist
+      (var/"lib/elasticsearch/#{cluster_name}").mkpath
+      (var/"log/elasticsearch").mkpath
+      ln_s etc/"elasticsearch", libexec/"config"
+      (var/"elasticsearch/plugins").mkpath
+      ln_s var/"elasticsearch/plugins", libexec/"plugins"
+    end
+  
+    def caveats
+      <<~EOS
+        Data:    #{var}/lib/elasticsearch/#{cluster_name}/
+        Logs:    #{var}/log/elasticsearch/#{cluster_name}.log
+        Plugins: #{var}/elasticsearch/plugins/
+        Config:  #{etc}/elasticsearch/
+      EOS
+    end
+  
+    # https://docs.brew.sh/Formula-Cookbook#service-block-methods
+    service do
+      run opt_bin/"elasticsearch"
+      working_dir var
+      environment_variables ES_JAVA_HOME: `/usr/libexec/java_home -v 17`.strip
```

Comment:
So Elasticsearch already knows what to do with `ES_JAVA_HOME` (as opposed to the previous `JAVA_HOME`?

Reply:
([Yes](https://github.com/search?q=repo%3Aelastic%2Felasticsearch%20ES_JAVA_HOME&type=code), I take it)

Reply:
`JAVA_HOME` also works, but generated deprecation warnings recommending `ES_JAVA_HOME` instead. Neither worked using the old, indirect `'/usr/libexec/java_home -v 17'` form. Instead I had to actually run the command during installation.